### PR TITLE
Entity Refactor

### PR DESCRIPTION
**Pull request by OR13**
Originally we tried a find replace solution: https://github.com/w3c-ccg/traceability-vocab/pull/387

This was rejected, for good reasons.

This PR proposes a more complicated alternative.

Context for the problem:

### 1. `oneOf` is poorly supported by OAS / JSON Schema processors

For example, this is what renders in JSON Schema form when you use oneOf:

Screen Shot 2022-04-27 at 11 52 36 AM

###  2. `oneOf` is not supported by the JSON-LD context builder

For example, this is the current context for `Entity`:

```
 "Entity": {
      "@id": "https://w3id.org/traceability#Entity",
      "@context": {}
    },
```

### 3. Entity as currently defined supports the following legal but probably not useful instances:

```
{
  "type: "Person",
  "firstName" : "Bob"
}
```

```
{
  "type: "Organization",
  "phone" : "555-555-5555"
}
```

Both of these are both valid JSON Schema Instances and Valid JSON-LD... yet very unhelpful to issuers or verifiers.

## Proposed Solution

This PR makes "Entity2" a JSON-LD Schema for an RDF Class, with required only attributes. 

These attributes must be present and must work for both "natural persons" and "organizations".

This allows for a verifier to handle entity under the intended semantics, and have confidence that all required fields are present... neither is functional in todays model.

Proof that the proposed solution works in JSON Schema Form:

Screen Shot 2022-04-27 at 12 46 46 PM

Proof that the proposed Solution works in JSON-LD:

https://v.jsld.org/BZqJG8rkribk4PN3WRQ2BFAYQsr4waKruGedwyD7TTjiziku3KiNJQ6ahDqtvCnJd4TXbADLW4Fbc3QCVtNrCe8sbJQWSD9VMGEQ6iFPdqo273EcycntwErBAFdbLrfR5i2PVjch3qm5L49KJmhra56yDKKK82xta3ZmNbJ2ej2tkJZPPrELWEKYaL6on9EHbhdJeaPbAQsNt1SU7TdQEsX6ffbxTmmHX8PqMoFCTtj5pzcXRssKzgE2S7LMohE2Z7okoL1YYve5cU5C

### CODE REVIEW

**Review by BenjaminMoe — APPROVED** [review]
I approve of this approach

**Comment by OR13** [comment]
If there is consensus on this approach, I can handle the extremely painful process of fixing this... everywhere... my guess is that it will yield a large number of issues where "Entity" was used when it should not have been used.

**Comment by BenjaminMoe** [comment]
My intended implementation of this would be merge `Organization` and `Person` into a new `Entity.yml` which would replace the existing `oneOf Entity.yml`. And then go back and re-generate all of the proofs. 

```yml
title: Entity
description: A merge of Person and Organization
type: object
  type:
    oneOf:
      - type: array
        items:
          type: string
          enum:
            - Entity
      - type: string
        const:
          - Entity
  entityType:
    description: The type of entity
    type: string
    enum:
      - Person
      - Organization
  name:
    title: Name
    description: Name of the organization.
    type: string
    $linkedData:
      term: name
      '@id': https://schema.org/name
  legalName:
    title: Legal Name
    description: The official name of the organization, i.e., the registered company name.
    type: string
    $linkedData:
      term: legalName
      '@id': https://schema.org/legalName
  firstName:
    title: First Name
    description: Person's first name.
    type: string
    $linkedData:
      term: firstName
      '@id': https://schema.org/givenName
  lastName:
    title: Last Name
    description: Person's last name.
    type: string
    $linkedData:
      term: lastName
      '@id': https://schema.org/familyName
  email:
    title: Person's Email Address
    description: Person's email address.
    type: string
    $linkedData:
      term: email
      '@id': https://schema.org/email
  phoneNumber:
    title: Phone Number
    description: Person's contact phone number.
    type: string
    $linkedData:
      term: phoneNumber
      '@id': https://schema.org/telephone
  worksFor:
    title: Works For
    description: Company which employs the person.
    $ref: ./Organization.yml
    $linkedData:
      term: worksFor
      '@id': https://schema.org/worksFor
  jobTitle:
    title: Job Title
    description: Person's job title.
    type: string
    $linkedData:
      term: jobTitle
      '@id': https://schema.org/jobTitle
  leiCode:
    title: Legal Entity Identifier Code
    description: Legal entity identifier code
    type: string
    $linkedData:
      term: leiCode
      '@id': https://schema.org/leiCode
  url:
    title: URL
    description: URL of the organization.
    type: string
    $linkedData:
      term: url
      '@id': https://schema.org/url
  description:
    title: Description
    description: Description of the company.
    type: string
    $linkedData:
      term: description
      '@id': https://schema.org/description
  globalLocationNumber:
    title: Global Location Number
    description: The GS1 GLN of the organization.
    type: string
    $linkedData:
      term: globalLocationNumber
      '@id': https://schema.org/globalLocationNumber
  address:
    title: Postal Address
    description: The postal address for an organization or place.
    $ref: ./PostalAddress.yml
    $linkedData:
      term: address
      '@id': https://schema.org/PostalAddress
  faxNumber:
    title: Phone Number
    description: Organization's fax number.
    type: string
    $linkedData:
      term: faxNumber
      '@id': https://schema.org/faxNumber
  brand:
    title: Brand
    description: Branding for this organization
    $ref: ./Brand.yml
    $linkedData:
      term: brand
      '@id': https://schema.org/Brand
  contactPoint:
    title: Contact Point
    description: Points of contact for this organization
    $ref: ./ContactPoint.yml
    $linkedData:
      term: contactPoint
      '@id': https://schema.org/ContactPoint
  taxId:
    title: Tax Identification Number
    description: >-
      The Tax / Fiscal ID of the organization or person, e.g., the TIN in the US
      or the CIF/NIF in Spain.
    type: string
    $linkedData:
      term: taxId
      '@id': https://schema.org/taxID
  iataCarrierCode:
    title: IATA Carrier Code
    description: Carrier's three-digit IATA airline code number
    type: string
    $linkedData:
      term: iataCarrierCode
      '@id': https://onerecord.iata.org/cargo/Company#airlineCode
```

**Comment by OR13** [comment]
@BenjaminMoe  Let's stick to discussing on issues, and keep PR feedback to change requests.

https://github.com/w3c-ccg/traceability-vocab/issues/395

File: docs/openapi/components/schemas/common/Entity2.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+$linkedData:
+  term: Entity2
+  '@id': https://w3id.org/traceability#Entity2
+title: Entity2
+description: A natural person or legal entity such as a corporation or government.
+type: object
+required:
+  - name
```
Comment (OR13) [inline docs/openapi/components/schemas/common/Entity2.yml:8]:
```suggestion
  - name
  - type
```

**Comment by TallTed** [comment]
> [@OR13] Let's stick to discussing on issues, and keep PR feedback to change requests.

This PR *starts* with a giant hunk of discussion, and does not say anything about "fixes issue #blah"!  Please, let's just close all the PRs (including this one) associated with this issue (which I think is the entire focus of #395, so all other related issues should *also* be closed), and have all discussion there (#395), until we settle on a viable path forward.

Then a new PR should be begun, to work on implementing that path.  Change requests (which should generally be small) will be appropriate on that PR.  Discussion of why that PR doesn't actually address that Issue might usually be considered to be appropriate as comments on that PR, but in this case should probably be made on the Issue with links/references to the PR.

**Comment by OR13** [comment]
Moving to discussion on #395

**Pull request closed without merging**